integreat_cms/static/src/js/regions/conditional-fields.ts

```diff
@@ -28,4 +28,33 @@ window.addEventListener('load', () => {
             }
         });
     }
-});
\ No newline at end of file
+    // add conditional logic to display options within the second timezone dropdown
+    const timezoneAreaDropdown = document.getElementById("id_timezone_area") as HTMLSelectElement;
+    const timezoneDropdown = document.getElementById("id_timezone") as HTMLSelectElement;
+        
+    if(timezoneDropdown && timezoneAreaDropdown) {
+        updateTimezoneAreaDropdown();
+        timezoneAreaDropdown.addEventListener('change', updateTimezoneAreaDropdown)
+    }
+});
+
+function updateTimezoneAreaDropdown(event?:Event) {
+    const timezoneAreaDropdown = document.getElementById("id_timezone_area") as HTMLSelectElement;
+    const timezoneDropdown = document.getElementById("id_timezone") as HTMLSelectElement;
+    let otherOptions;
+    let selectedOptions;
+    if (event !== undefined) {
+        timezoneDropdown.value = '';
+    }
+    if (timezoneAreaDropdown.value === '') {
+        selectedOptions = [];
+        otherOptions = timezoneDropdown.querySelectorAll('option');
+    } else {
+        otherOptions = timezoneDropdown.querySelectorAll('option:not([value*="' + timezoneAreaDropdown.value + '"])');
+        selectedOptions = timezoneDropdown.querySelectorAll('option[value*="' + timezoneAreaDropdown.value + '"]');
+    }
+    otherOptions.forEach((option) => option.classList.add('hidden'));
+    selectedOptions.forEach((option) => option.classList.remove('hidden'));
+
+    timezoneDropdown.querySelector('option[value=""]').classList.remove('hidden');
+}
```

Comment:
```suggestion
function updateTimezoneAreaDropdown(event?: Event) {
    const timezoneAreaDropdown = document.getElementById("id_timezone_area") as HTMLSelectElement;
    const timezoneDropdown = document.getElementById("id_timezone") as HTMLSelectElement;
    // Reset timezone value when timezone region is changed
    if (event !== undefined) {
        timezoneDropdown.value = "";
    }
    // Hide all sub-timzones of other areas
    timezoneDropdown
        .querySelectorAll('option:not([value*="' + timezoneAreaDropdown.value + '"],[value=""])')
        .forEach((option) => option.classList.add("hidden"));
    // Show all sub-timezones of the selected area
    timezoneDropdown
        .querySelectorAll('option[value*="' + timezoneAreaDropdown.value + '"]')
        .forEach((option) => option.classList.remove("hidden"));
}

```
I found a way to simplify this function - it turns out we don't need the if clause...